lib/cartopy/crs.py

```diff
@@ -156,6 +156,41 @@ def _determine_longitude_bounds(self, central_longitude):
             minlon += epsilon
         return minlon, maxlon
 
+    def _repr_svg_(self):
+        """
+        Make a visual representation of the projection and return it as an
+        html element.
+
+        """
+        # Imports.
+        try:
+            # As matplotlib is not a core cartopy dependency, don't error
+            # if it's not available, but fall back to the default repr.
+            import matplotlib.pyplot as plt
+        except ImportError:
+            pass
+        else:
+            if six.PY2:
+                from StringIO import StringIO
+            else:
+                from io import StringIO
+            # Produce a visual repr of the Projection instance.
+            fig, ax = plt.subplots(figsize=(5, 3),
+                                   subplot_kw={'projection': self})
+            ax.set_global()
+            ax.coastlines('110m')
+            ax.gridlines()
+            # "Save" to a string buffer.
+            buf = StringIO()
+            fig.savefig(buf, format='svg')
+            plt.close(fig)
+            # "Rewind" the buffer to the start and return it as an svg string.
+            buf.seek(0)
+            return buf.read()
+        finally:
```

Comment:
Can you take out the `finally` clause? Because it’s in the `finally`, that code runs for *every* invocation of `_repr_svg`. Also, just calling `repr` wouldn’t be enough, you need to return it.

Reply:
In addition, returning a string as an SVG isn't going to end well I don't think.